examples/quickstart-sklearn-tabular/sklearnexample/server_app.py

```diff
@@ -0,0 +1,67 @@
+from typing import List, Tuple, Dict
+
+from flwr.common import Metrics, Scalar, Context
+from flwr.server import ServerAppComponents, ServerConfig, ServerApp
+from flwr.server.strategy import FedAvg
+
+from example.task import set_initial_params
+
+
+def weighted_average(metrics: List[Tuple[int, Metrics]]) -> Dict[str, Scalar]:
+    """Compute weighted average.
+
+    It is a generic implementation that averages only over floats and ints and drops the
+    other data types of the Metrics.
+    """
+    # num_samples_list can represent the number of samples
+    # or the number of batches depending on the client
+    num_samples_list = [n_batches for n_batches, _ in metrics]
+    num_samples_sum = sum(num_samples_list)
+    metrics_lists: Dict[str, List[float]] = {}
+    for num_samples, all_metrics_dict in metrics:
+        #  Calculate each metric one by one
+        for single_metric, value in all_metrics_dict.items():
+            if isinstance(value, (float, int)):
+                metrics_lists[single_metric] = []
+        # Just one iteration needed to initialize the keywords
+        break
+
+    for num_samples, all_metrics_dict in metrics:
+        # Calculate each metric one by one
+        for single_metric, value in all_metrics_dict.items():
+            # Add weighted metric
+            if isinstance(value, (float, int)):
+                metrics_lists[single_metric].append(float(num_samples * value))
+
+    weighted_metrics: Dict[str, Scalar] = {}
+    for metric_name, metric_values in metrics_lists.items():
+        weighted_metrics[metric_name] = sum(metric_values) / num_samples_sum
+
+    return weighted_metrics
+
+
+def server_fn(context: Context) -> ServerAppComponents:
+    """Construct components that set the ServerApp behaviour.
+
+    You can use settings in `context.run_config` to parameterize the
+    construction of all elements (e.g the strategy or the number of rounds)
+    wrapped in the returned ServerAppComponents object.
+    """
+
+    # Define the strategy
+    min_available_clients = context.run_config["min-available-clients"]
+    strategy = FedAvg(
+        min_available_clients=min_available_clients,
+        fit_metrics_aggregation_fn=weighted_average,
+        evaluate_metrics_aggregation_fn=weighted_average,
```

Comment:
When running the app i'm only seeing `0.0` in `test_accuracy`...

Reply:
Did it happen once you changed the dataset? For me, it works with the `hitorilabs/iris`

Reply:
yes, maybe that's why